http-daemon/src/main/java/org/sonarlint/daemon/SourceRepository.java

```diff
@@ -20,98 +20,64 @@
 package org.sonarlint.daemon;
 
 import java.io.IOException;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.UUID;
-import org.apache.http.HttpEntity;
-import org.apache.http.HttpHost;
-import org.apache.http.entity.StringEntity;
-import org.apache.http.message.BasicHeader;
-import org.elasticsearch.action.get.GetRequest;
-import org.elasticsearch.action.get.GetResponse;
-import org.elasticsearch.action.index.IndexRequest;
-import org.elasticsearch.client.RestClient;
-import org.elasticsearch.client.RestHighLevelClient;
-
-import static org.apache.http.entity.ContentType.APPLICATION_JSON;
-import static org.apache.http.protocol.HTTP.CONTENT_TYPE;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
 
 public class SourceRepository {
 
-  private RestClient lowLevelClient;
-  private RestHighLevelClient highLevelClient;
+  private Path storeDir;
+
+  public void init(Path workDir) {
+    if (storeDir == null) {
+      storeDir = workDir.resolve("store");
+      try {
+        Files.createDirectories(storeDir);
+      } catch (IOException e) {
+        e.printStackTrace();
+        throw new IllegalStateException("Cannot create store");
+      }
+      Logger.getLogger(Backend.class.getName()).warning(() -> "storeDir: " + storeDir.toAbsolutePath());
+    }
+  }
 
-  public String load(String uuid) {
-    init();
+  public String load(String storedAs) {
+    RandomString.check(storedAs);
+    List<String> content;
     try {
-      GetResponse response = highLevelClient.get(new GetRequest("files", "file", uuid).storedFields("code"));
-      return (String) response.getField("code").getValue();
+      content = Files.readAllLines(getFile(storedAs));
     } catch (IOException e) {
-      throw new IllegalStateException("Cannot load "+uuid, e);
+      e.printStackTrace();
+      throw new IllegalStateException("Cannot read file");
     }
+    return content.stream().collect(Collectors.joining("\n"));
```

Comment:
You could use `new String(Files.readAllBytes(path), charset)` instead of reading a list of lines and joining.

Reply:
Done.